### PR TITLE
core(route): Migrate to go_router.

### LINKED ISSUE #285: [SUGGESTION]: Replace auto_route with go_router
**Is your feature request related to a problem? Please describe.**
We want to change to go_router so that we can get rid of auto_route code generation.

Replace all routes with go_router

@wizlif Please elaborate as much as possible.

### PR DESCRIPTION
**Pull request by wizlif**
Complete migration from auto_route to go_router.
Much shorter build runs (Only freezed 😄 )

Will still need some extensive testing

Closes #285

### CODE REVIEW

File: lib/infrastructure/core/settings_repository.dart [hunk 1]
```diff
@@ -46,14 +48,17 @@ class SettingsRepository implements ISettingsRepository, Disposable {
 
   @override
   Future<bool> getWasUserOnboarded() async {
+    if (_isOnBoarded != null) return _isOnBoarded ?? false;
     await _initCompleter.future;
-    return _prefs.getBool(_prefsKeyWasUserOnboarded) ?? false;
+    _isOnBoarded = _prefs.getBool(_prefsKeyWasUserOnboarded) ?? false;
+    return _isOnBoarded ?? false;
```
Comment (Xazin) [inline lib/infrastructure/core/settings_repository.dart:54]: Cannot be null

Comment (Xazin) [inline lib/infrastructure/core/settings_repository.dart:54]:
```suggestion
    return _isOnBoarded!;
```

**Review by Xazin — APPROVED** [review]

**Comment by wizlif** [comment]
There is still a trending issue on go_router shell route about maintaining page state. I will not merge till there is a fix for it. The P.R is still in review and hopefully won't take long.

https://github.com/flutter/flutter/issues/99124
https://github.com/flutter/packages/pull/2650

**Comment by Xazin** [comment]
> There is still a trending issue on go_router shell route about maintaining page state. I will not merge till there is a fix for it. The P.R is still in review and hopefully won't take long.
> 
> [flutter/flutter#99124](https://github.com/flutter/flutter/issues/99124) [flutter/packages#2650](https://github.com/flutter/packages/pull/2650)

Alright sounds good, let's try to rebase this branch once in a while or similar, to make the merge easier later.

**Comment by Xazin** [comment]
@wizlif Version 7.1.0 of Go Router resolves the issues we have been waiting for :tada:

**Comment by wizlif** [comment]
Let me do a fresh version of this, feels like a rebase will be crazy

**Pull request closed without merging**